Source/WebCore/dom/Element.cpp

```diff
@@ -2033,7 +2041,36 @@ void Element::setElementAttribute(const QualifiedName& attributeName, Element* e
     explicitlySetAttrElementsMap().set(attributeName, Vector<WeakPtr<Element>> { element });
 }
 
-std::optional<Vector<RefPtr<Element>>> Element::getElementsArrayAttribute(const QualifiedName& attributeName) const
+JSC::JSValue Element::getElementsArrayAttribute(const QualifiedName& attributeName, JSC::JSGlobalObject& lexicalGlobalObject, WebCore::JSDOMGlobalObject& globalObject)
+{
+    JSValueInWrappedObject* cachedValue = nullptr;
+    if (auto* cachedMap = cachedAttrAssociatedElementsMapIfExists()) {
+        if (auto it = cachedMap->find(attributeName); it != cachedMap->end())
+            cachedValue = it->value.get();
+    }
+
+    std::optional<Vector<RefPtr<Element>>> elements = getElementsArrayAttributeInternal(attributeName);
+    if (!elements) {
+        if (cachedValue)
+            cachedValue->clear();
+        return JSC::jsNull();
+    }
+
+    JSC::JSValue newValue = JSConverter<IDLFrozenArray<IDLInterface<Element>>>::convert(lexicalGlobalObject, globalObject, elements.value());
+    if (cachedValue) {
+        if (!cachedValue->getValue().isUndefinedOrNull()) {
+            Vector<RefPtr<Element>> cachedElements = Converter<IDLFrozenArray<IDLInterface<Element>>>::convert(lexicalGlobalObject, cachedValue->getValue());
```

Comment:
Is there a way to invalidate the cache when the attribute changes instead of doing this somewhat expensive-looking work every time we access the attribute?